gcc 11 compiler error

```
src/vt/trace/trace_common.h:55:29: error:
            'size_t' does not name a type
      55 | using TraceEntryIDType    = size_t; // uint32_t, maybe..
           |                             ^~~~~~
```

Likely this should be fixed by  including the correct header for size_t

--- a/src/vt/trace/trace_common.h
+++ b/src/vt/trace/trace_common.h
@@ -47,6 +47,7 @@
 #include "vt/configs/types/types_type.h"
 
 #include <cstdint>
+#include <cstddef>
 
 namespace vt { namespace trace {
 

